File: prak3_java/Pc.java
public class Pc {
    
    private Processor processor;
    private Disk disk;
    private Ram ram;
    private int totalPrice;

    //konstruktor
    public Pc() {
    }

    public Pc(Processor processor, Disk disk, Ram ram) {
        this.processor = processor;
        this.disk = disk;
        this.ram = ram;
        this.totalPrice = processor.getPrice() + disk.getPrice() + ram.getPrice();
    }

    //getter setter
    public Processor getProcessor() {
        return processor;
    }

    public void setProcessor(Processor processor) {
        this.processor = processor;
    }

    public Disk getDisk() {
        return disk;
    }

    public void setDisk(Disk disk) {
        this.disk = disk;
    }

    public Ram getRam() {
        return ram;
    }

    public void setRam(Ram ram) {
        this.ram = ram;
    }

    public int getTotalPrice() {
        return totalPrice;
    }

    public void updateTotalPrice(){
        this.totalPrice = processor.getPrice() + disk.getPrice() + ram.getPrice();
    }

    //methods
    public void show(){
        System.out.println("Processor");
        System.out.println("- Nama : " + processor.getName());
        System.out.println("- Price : " + processor.getPrice());
        System.out.println("Disk");
        System.out.println("- Type : " + disk.getType());
        System.out.println("- Capacity : " + disk.getCapacity());
        System.out.println("- Price : " + disk.getPrice());
        System.out.println("RAM");
        System.out.println("- Capacity : " + ram.getCapacity());
        System.out.println("- Price : " + ram.getPrice());
        System.out.println("Total Price : " + getTotalPrice());
    }

}

File: prak3_c++/Pc.cpp
#include <iostream>

#include "Processor.cpp"
#include "Disk.cpp"
#include "Ram.cpp"

using std::cout;
using std::endl;

class Pc {
    private:
        Processor processor;
        Disk disk;
        Ram ram;
        int totalPrice;
        
    public:
        //konstruktor
        Pc(){
            totalPrice = 0;
        }

        Pc(Processor pr, Disk di, Ram ra){
            processor = pr;
            disk = di;
            ram = ra;
            totalPrice = pr.getPrice() + di.getPrice() + ra.getPrice();
        }


        //setter
        void setProcessor(Processor pr){
            processor = pr;
        }

        void setDisk(Disk di){
            disk = di;
        }

        void setRam(Ram ra){
            ram = ra;
        }

        //getter
        Processor getProcessor(){
            return processor;
        }

        Disk getDisk(){
            return disk;
        }

        Ram getRam(){
            return ram;
        }

        int getTotalPrice(){
            return totalPrice;
        }

        //methods
        void updateTotalPrice(){
            totalPrice = processor.getPrice() + disk.getPrice() + ram.getPrice();
        }

        void show(){
            cout << "Processor" << endl;
            cout << "- Name : " << processor.getName() << endl;
            cout << "- Price : " << processor.getPrice() << endl;
            cout << "Disk" << endl;
            cout << "- Type : " << disk.getType() << endl;
            cout << "- Capacity : " << disk.getCapacity() << endl;
            cout << "- Price : " << disk.getPrice() << endl;
            cout << "RAM" << endl;
            cout << "- Capacity : " << ram.getCapacity() << endl;
            cout << "- Price : " << ram.getPrice() << endl;
            cout << "Total Price : " << getTotalPrice() << endl;
        }


        //destruktor
        ~Pc(){
        }

};
